src/test/java/at/ac/fhcampuswien/fhmdb/HomeControllerTest.java

```diff
@@ -142,6 +165,50 @@ void tearDown() {
         movieAPIServiceMockedStatic.close();
     }
 
+    // Streams test
+    @Nested
+    class StreamsTest{
+        @Test
+        void testGetMostPopularActor() {
+
+            String mostPopularActor = homeController.getMostPopularActor(streamTestMovies);
+
+            assertEquals("Zachary Quinto", mostPopularActor);
+        }
+
+        @Test
+        void testGetLongestMovieTitle() {
+
+            int longestMovieTitle = homeController.getLongestMovieTitle(streamTestMovies);
+
+            assertEquals(17, longestMovieTitle);
+        }
+
+        @Test
+        void testCountMoviesFrom() {
+
+            long moviesFromJohnMcTiernan = homeController.countMoviesFrom(streamTestMovies, "John McTiernan");
+            long moviesFromNonExistentDirector = homeController.countMoviesFrom(streamTestMovies, "Non-existent Director");
+
+            assertEquals(1, moviesFromJohnMcTiernan);
+            assertEquals(0, moviesFromNonExistentDirector);
+        }
+
+        @Test
+        void testGetMoviesBetweenYears() {
+
+            List<Movie> moviesBetween1988And1998 = homeController.getMoviesBetweenYears(streamTestMovies, 1988, 1998);
+            List<Movie> moviesBetween2000And2010 = homeController.getMoviesBetweenYears(streamTestMovies, 2000, 2010);
+            List<Movie> moviesBetween2020And2030 = homeController.getMoviesBetweenYears(streamTestMovies, 2020, 2030);
+
+            assertEquals(3, moviesBetween1988And1998.size());
+            assertEquals(2, moviesBetween2000And2010.size());
+            assertEquals(0, moviesBetween2020And2030.size());
+        }
```

Comment:
1. Use snake case for method naming for improved readability (mentioned in the videos about testing)
2. Remove the word `test` in front of all methods for improved readability (mentioned in the videos about testing)
3. Make comments for `//Arrange //Act //Assert `sections. If `//Arrange` is in the beforeAll method write `//Arrange is in beforeAll`
4. I would stick to the naming convention we already got with the @Nested annotation where the class is called like the method which is tested and the method name of the test describes what the test does. (This was also suggested in the videos about testing)

So you would have 4 @Nested classes, one for each method you wrote.

The first one would look something like this:

@Nested
class GetMostPopularActor{
@Test
void Gets_the_right_actor(){...}
.
.
.
}

Reply:
Implemente recommendations